testrunner/src/mill/testrunner/TestRunnerUtils.scala

```diff
@@ -208,6 +210,145 @@ import scala.jdk.CollectionConverters.IteratorHasAsScala
     (doneMessage, results.toSeq)
   }
 
+  def stealTasks(
+      testClasses: Seq[(Class[?], Fingerprint)],
+      testReporter: TestReporter,
+      runner: Runner,
+      stealFolder: os.Path,
+      selectorFolder: os.Path
+  )(implicit ctx: Ctx.Log): (String, Iterator[TestResult]) = {
+    val events = new ConcurrentLinkedQueue[Event]()
+    val doneMessage = {
+
+      val taskQueue = mutable.Queue.empty[Task]
```

Comment:
With the changes suggested for `stealFromSelectorFolder` and `stealTaskLoop`, we should be able to remove this mutable `taskQueue` as well